### PR TITLE
Fix typings for <details>

### PR DESCRIPTION
**Pull request by atk**
Typing details as DetailsHTMLAttributes<HTMLElement> will stop the open property from working correctly in typescript.

### CODE REVIEW

File: packages/dom-expressions/src/jsx.d.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export namespace JSX {
   interface ArrayElement extends Array<Element> {}
   interface FunctionElement {
     (): Element;
-  }
```
Comment (ryansolid) [inline packages/dom-expressions/src/jsx.d.ts:22]: Was this removal intentional?

Comment (atk) [reply packages/dom-expressions/src/jsx.d.ts:22]: No, it wasn't.

**Comment by ryansolid** [comment]
Thank you
